service-base/src/main/java/org/eclipse/hono/service/http/AbstractVertxHttpServer.java

```diff
@@ -0,0 +1,549 @@
+/**
+ * Copyright (c) 2016, 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+
+package org.eclipse.hono.service.http;
+
+import io.vertx.core.CompositeFuture;
+import io.vertx.core.Future;
+import io.vertx.core.http.*;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.Router;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.BodyHandler;
+import org.eclipse.hono.config.ServiceConfigProperties;
+import org.eclipse.hono.service.AbstractProtocolAdapterBase;
+import org.eclipse.hono.util.Constants;
+import org.eclipse.hono.util.JwtHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static java.net.HttpURLConnection.HTTP_BAD_REQUEST;
+import static java.net.HttpURLConnection.HTTP_INTERNAL_ERROR;
+import static java.net.HttpURLConnection.HTTP_UNAVAILABLE;
+
+/**
+ * Abstract VertX based HTTP server.
+ *
+ * @param <T> The type of configuration properties used by this service.
+ */
+public abstract class AbstractVertxHttpServer<T extends ServiceConfigProperties> extends AbstractProtocolAdapterBase<T> {
+    /**
+     * The <em>application/json</em> content type.
+     */
+    protected static final String CONTENT_TYPE_JSON = "application/json";
```

Comment:
I think to subclass AbstractProtocolAdapterBase is not appropriate here, since an abstract http server class must not deal with HonoClients (like the AbstractProtocolAdapterBase class does).
Can you please rewrite this part so no protocol adapter specifics (especially no HonoClients) are inherited into this class anymore?